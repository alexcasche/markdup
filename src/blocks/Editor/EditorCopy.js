import styled from 'styled-components'
import { tint } from 'polished'
import { applyStyleModifiers } from 'styled-components-modifiers'

const MODIFIERS = {
  visible: ({ theme }) => `
    opacity: 1;
  `,
}

const EditorCopy = styled.div`
  position: absolute;
  top: 42px;
  right: -20px;
  width: 160px;
  pointer-events: none;
  padding: 8px 16px;
  background-color: ${props => tint(0.85, props.theme.colors.black)};
  border-radius: 2px;
  color: ${props => tint(0.95, props.theme.colors.grey)};
  font-size: 14px;
  text-transform: capitalize;
  opacity: 0;
  transition: opacity 0.1s ease-in-out;
  svg {
    position: absolute;
    right: 26px;
    top: -18px;
    height: 30px;
    color: ${props => tint(0.85, props.theme.colors.black)};
  }
  ${applyStyleModifiers(MODIFIERS)};
`

export default EditorCopy
