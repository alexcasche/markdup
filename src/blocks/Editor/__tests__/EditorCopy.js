import EditorCopy from '../EditorCopy'

describe('EditorCopy Element', () => {
  it('renders correctly', () => {
    const tree = render(<EditorCopy modifiers={[]} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

it('adds expanded modifiers', () => {
  const tree = render(<EditorCopy modifiers={['visible']} />).toJSON()
  expect(tree).toMatchSnapshot()
  expect(tree).toHaveStyleRule('opacity', '1')
})
