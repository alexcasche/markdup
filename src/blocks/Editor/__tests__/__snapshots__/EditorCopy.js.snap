// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditorCopy Element renders correctly 1`] = `
.c0 {
  position: absolute;
  top: 42px;
  right: -20px;
  width: 160px;
  pointer-events: none;
  padding: 8px 16px;
  background-color: #3e4042;
  border-radius: 2px;
  color: #afb1b6;
  font-size: 14px;
  text-transform: capitalize;
  opacity: 0;
  -webkit-transition: opacity .1s ease-in-out;
  transition: opacity .1s ease-in-out;
}

.c0 svg {
  position: absolute;
  right: 26px;
  top: -18px;
  height: 30px;
  color: #3e4042;
}

<div
  className="c0"
/>
`;

exports[`adds expanded modifiers 1`] = `
.c0 {
  position: absolute;
  top: 42px;
  right: -20px;
  width: 160px;
  pointer-events: none;
  padding: 8px 16px;
  background-color: #3e4042;
  border-radius: 2px;
  color: #afb1b6;
  font-size: 14px;
  text-transform: capitalize;
  opacity: 0;
  -webkit-transition: opacity .1s ease-in-out;
  transition: opacity .1s ease-in-out;
  opacity: 1;
}

.c0 svg {
  position: absolute;
  right: 26px;
  top: -18px;
  height: 30px;
  color: #3e4042;
}

<div
  className="c0"
/>
`;
